src/components/input/inputContainer/inputContainer.api.ts

```diff
@@ -0,0 +1,56 @@
+import type { InputHTMLAttributes, ReactNode } from 'react';
+
+export type InputVariant = 'default' | 'warning' | 'critical';
+
+export interface IInputContainerProps {
+    /**
+     * Label of the input.
+     */
+    label?: string;
+    /**
+     * Variant of the input.
+     * @default default
+     */
+    variant?: InputVariant;
+    /**
+     * Help text displayed above the input.
+     */
+    helpText?: string;
+    /**
+     * Displays the optional tag when set to true.
+     */
+    isOptional?: boolean;
+    /**
+     * Displays the input as disabled when set to true.
+     */
+    isDisabled?: boolean;
```

Comment:
It is to follow a shared naming convention for such state / variant properties (like `isDisabled`, `isLoading`, `isOptional` and so on). I would prefer to use `isDisabled` for `disabled` instead of changing the others to `loading` or `optional`, having the `is` feels a bit more descriptive. What do you think?

Reply:
I see. The suggestion was so that disabled could be masqueraded as the usual out of the box disabled attribute.